### PR TITLE
feat: set title of the tab dynamically.

### PR DESCRIPTION
**Pull request by saicharan-zemoso**
fixes issue: [34260](https://github.com/appsmithorg/appsmith/issues/34260)

added the code which can abilitate the user to set the title of the tab dynamically.

screen shot of the working application:
Screenshot from 2024-09-12 14-18-03

here is the video of the application:

https://github.com/user-attachments/assets/8bad9c78-2afe-4924-8bb9-c2c05ddbd0e5

## Summary by CodeRabbit

- **New Features**
	- Introduced a dynamic naming feature for tabs in the `TabsWidget`, allowing users to customize tab labels.
	- Added a test suite to verify the functionality of dynamic tab naming and user interactions.

- **Bug Fixes**
	- Enhanced validation for tab label input to ensure it accepts only text.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Tab/Tabs_dynamic_tab_name.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  agHelper,
+  locators,
+  propPane,
+} from "../../../../../support/Objects/ObjectsCore";
+import EditorNavigation, {
+  EntityType,
+} from "../../../../../support/Pages/EditorNavigation";
+
+describe(
+  "Test to check if user name the tab dynamically",
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Tab/Tabs_dynamic_tab_name.ts:11]: Can we test this with unit testing please?

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Tab/Tabs_dynamic_tab_name.ts (hunk 1 above)
Comment (saicharan-zemoso) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Tab/Tabs_dynamic_tab_name.ts:11]: Ok sure will work on that.

Comment (saicharan-zemoso) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Tab/Tabs_dynamic_tab_name.ts:11]: Hello @sagar-qa007 ,

I don't know what to include in the unit test case.
As
I have added the cypress test should I try to replicate it using the react testing library??.

if Yes , can you provide any reference test file or test case in which this as done??

**Comment by saicharan-zemoso** [comment]
Hello @rahulbarwal , could you review this PR if you have some band width.

As the original component did not have any unit test case. I did't know what to add in the test as the change is only made in pane configurations.

If this PR needs a unit test can you suggest what to include or could you provide any reference unit test so that I can work on it.

It would be very helpful.
Thank you.

**Pull request closed without merging**